meilisearch/tests/documents/errors.rs

```diff
@@ -82,6 +82,110 @@ async fn get_all_documents_bad_limit() {
     "###);
 }
 
+#[actix_rt::test]
+async fn get_all_documents_bad_filter() {
+    let server = Server::new().await;
+    let index = server.index("test");
+
+    // Since the filter can't be parsed automatically by deserr, we have the wrong error message
+    // if the index does not exists
```

Comment:
What is the expected error message? Why is the one below wrong?

```suggestion
    // if the index does not exist: we could expect to get an error message about the invalid filter before
    // the existence of the index is checked, but it is not the case.
```

going out on a limb as to what the reason might be ☝️

Reply:
The expected error messages are the ones we get once the index is created.
The ones right below are « wrong » because deserr can't parse the filter; thus, we have to do it **after** the index existence has been checked instead of doing it while deserializing the payload as we do for every other field.